diff --git a/src/JSONPath.php b/src/JSONPath.php
index f57f102..8ce2c16 100644
--- a/src/JSONPath.php
+++ b/src/JSONPath.php
@@ -210,6 +210,7 @@ class JSONPath implements ArrayAccess, Iterator, JsonSerializable, Countable
     /**
      * @inheritDoc
      */
+    #[\ReturnTypeWillChange]
     public function offsetGet($offset)
     {
         $value = AccessHelper::getValue($this->data, $offset);
@@ -250,6 +251,7 @@ class JSONPath implements ArrayAccess, Iterator, JsonSerializable, Countable
     /**
      * @inheritDoc
      */
+    #[\ReturnTypeWillChange]
     public function current()
     {
         $value = current($this->data);
@@ -268,6 +270,7 @@ class JSONPath implements ArrayAccess, Iterator, JsonSerializable, Countable
     /**
      * @inheritDoc
      */
+    #[\ReturnTypeWillChange]
     public function key()
     {
         return key($this->data);
